aea/protocols/generator/validate.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
```

Comment:
This module has code to validate a protocol specification. It crucially no longer does validation on Pythonic conversion of specification fields. All validation is done purely on the specification. 

This means it can be reused just before converting specification to other languages.

Reply:
Currently it has the existing validation logic. Other format checking logic (an outstanding generator issue) will be added in this module.